### PR TITLE
Add jshint linting to .travis.yml

### LINKED ISSUE #173: Should Travis runs include a JavaScript linting step?
This codebase is currently 81% Python and 18% JavaScript.  Travis currently runs flake8 to lint the Python code.  Should .travis.yml be modified to run JSHint or similar to keep the JS code in line?

### PR DESCRIPTION
**Pull request by cclauss**
Fixes #173

@chicco785 Perhaps you can give me some hints on
* https://travis-ci.org/Fiware/dataModels/jobs/290872645#L470

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,13 @@ python:
   - 3.6
 install:
   - pip install flake8
+  - npm install jshint
```
Comment (chicco785) [inline .travis.yml]: try with ``npm install -g jshint``

**Comment by chicco785** [comment]
see my comment to the installation, i believe that without the ``-g`` option, you install only the node library in the path where you execute the command (i.e. locally). using ``-g`` option you install it for all users and attach it to the global path.
https://docs.npmjs.com/getting-started/installing-npm-packages-globally

**Comment by cclauss** [comment]
Nice work @chicco785 ...  This is ready for review and to be merged.   I put in "warning only" mode which we can change once the 145 issues are fixed.  https://travis-ci.org/Fiware/dataModels/jobs/291467107#L576

File: .travis.yml [hunk 2]
```diff
@@ -5,11 +5,13 @@ python:
   - 3.6
 install:
   - pip install flake8
+  - npm install -g jshint
 before_script:
   # stop the build if there are Python syntax errors, PEP8 violations, undefined names
   - flake8 . --count --select=E,F821,F822,F823 --max-line-length=127 --show-source --statistics
   # exit-zero treats all errors as warnings.  GitHub editor is 127 chars wide
   - flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+  - jshint . || true  # treats all errors as warnings
```
Comment (jmcanterafonseca) [inline .travis.yml]: I would prefer to have the build failing rather than pass with warning so I would remove the || true part

**Comment by jmcanterafonseca** [comment]
LGTM, thanks. Now we need to fix the errors :)

**Pull request merged**